internal/access/scim.go

```diff
@@ -0,0 +1,20 @@
+package access
+
+import (
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+func ListProviderUsers(c *gin.Context, p *data.SCIMParameters) ([]models.ProviderUser, error) {
+	// this can only be run by an access key issued for an identity provider
+	ctx := GetRequestContext(c)
+	users, err := data.ListProviderUsers(ctx.DBTxn, ctx.Authenticated.AccessKey.IssuedFor, p)
```

Comment:
```suggestion
	ctx := GetRequestContext(c)
	// IssusedFor will match no providers if called with a regular access key. When called with
	// a SCIM access key it will be the provider ID. This effectively restricts this endpoint to
	// only SCIM access keys.
	users, err := data.ListProviderUsers(ctx.DBTxn, ctx.Authenticated.AccessKey.IssuedFor, p)
```

Maybe something along these lines to clarify how this is being authorized?